pkg/types/chains/evm/evm.go

```diff
@@ -0,0 +1,104 @@
+package evm
+
+import (
+	"math/big"
+	"time"
+)
+
+// represents evm-style address 40-character hexadecimal string prefixed by 0x making it 42 characters total
+// lower case, upper case or checksummed case (EIP-55) are allowed.
```

Comment:
So it is OK to be permissible with casing on input params since we can translate them to bytes anyways, but we need to be careful about returning things in mixed formats, because without a canonical format equality comparison breaks downs and they can't be used as map keys, etc.